terraform/modules/ecs/ecs.tf

```diff
@@ -0,0 +1,231 @@
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  task_role_arn            = aws_iam_role.task.arn
+
+  container_definitions = jsonencode([
+    {
+      name              = "${var.prefix}-ecs-db-migrate-container"
+      command           = ["bundle", "exec", "rake", "db:migrate"]
+      cpu               = var.ecs_task_cpu
+      environment       = var.ecs_environment_variables
+      essential         = true
+      image             = var.ecr_repository_url
+      memoryReservation = var.ecs_task_memory * 0.75
+      user              = "nonroot"
+
+      logConfiguration = {
+        logDriver = "awslogs"
+        options = {
+          awslogs-group         = aws_cloudwatch_log_group.main.id
+          awslogs-region        = "eu-west-2"
+          awslogs-stream-prefix = var.prefix
+        }
+      }
+
+      portMappings = [
+        {
+          protocol      = "tcp"
+          containerPort = 8080
+          hostPort      = 8080
+        }
+      ]
+
+      secrets = [for key, value in var.ecs_parameters : {
+        name      = key
+        valueFrom = value
+      }]
+    }
+  ])
+
+  runtime_platform {
+    operating_system_family = "LINUX"
+    cpu_architecture        = "X86_64"
+  }
+}
+
+resource "aws_ecs_task_definition" "db_seed" {
+  #checkov:skip=CKV_AWS_336:using readonlyRootFilesystem to true breaks the app, as it needs to write to app/tmp/pids for example
+  family                   = "${var.prefix}-ecs-db-seed-task"
+  cpu                      = var.ecs_task_cpu
+  execution_role_arn       = aws_iam_role.task_execution.arn
+  memory                   = var.ecs_task_memory #MiB
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  task_role_arn            = aws_iam_role.task.arn
+
+  container_definitions = jsonencode([
+    {
+      name              = "${var.prefix}-ecs-db-setup-container"
+      command           = ["bundle", "exec", "rake", "db:seed"]
+      cpu               = var.ecs_task_cpu
+      environment       = var.ecs_environment_variables
+      essential         = true
+      image             = var.ecr_repository_url
+      memoryReservation = var.ecs_task_memory * 0.75
+      user              = "nonroot"
+
+      logConfiguration = {
+        logDriver = "awslogs"
+        options = {
+          awslogs-group         = aws_cloudwatch_log_group.main.id
+          awslogs-region        = "eu-west-2"
+          awslogs-stream-prefix = var.prefix
+        }
+      }
+
+      portMappings = [
+        {
+          protocol      = "tcp"
+          containerPort = 8080
+          hostPort      = 8080
+        }
+      ]
+
+      secrets = [for key, value in var.ecs_parameters : {
+        name      = key
+        valueFrom = value
+      }]
+    }
+  ])
+
+  runtime_platform {
+    operating_system_family = "LINUX"
+    cpu_architecture        = "X86_64"
+  }
+}
+
+resource "aws_ecs_task_definition" "db_setup" {

```

Comment:
One of the reasons the app wasn't running at first was because we have to setup the db first. So I ran this task definition on the ECS, and then was able to run the app. So I've added this here so we can reuse it in future (e.g when using ECS on dev and prod for the first time)